Build with gcov is broken

## Required information

**Operating system:**
Ubuntu 20.04 LTS

**Compiler version:**
GCC 7.5.0

**Observed result or behaviour:**
Build with coverage enabled fails.

**Expected result or behaviour:**
Build works with coverage.

**Conditions where it occurred / Performed steps:**
Build broke when PR #487 was merged.
Build command: `./tools/iceoryx_build_test.sh build-strict clean debug test -c all`
Error:
```
/usr/bin/ld: CMakeFiles/test_stress_sofi.dir/stresstests/test_stress_sofi.cpp.o: in function `_GLOBAL__sub_I_00100_0__ZN51SoFiStress_SimultaneouslyPushAndPopOnEmptySoFi_Test10test_info_E':
/home/htm2abt/repo/ice_github/iceoryx_utils/test/stresstests/test_stress_sofi.cpp:552: undefined reference to `__gcov_init'
/usr/bin/ld: CMakeFiles/test_stress_sofi.dir/stresstests/test_stress_sofi.cpp.o: in function `_GLOBAL__sub_D_00100_1__ZN51SoFiStress_SimultaneouslyPushAndPopOnEmptySoFi_Test10test_info_E':
/home/htm2abt/repo/ice_github/iceoryx_utils/test/stresstests/test_stress_sofi.cpp:552: undefined reference to `__gcov_exit'
/usr/bin/ld: CMakeFiles/test_stress_sofi.dir/stresstests/test_stress_sofi.cpp.o:(.data.rel+0x20): undefined reference to `__gcov_merge_add'
collect2: error: ld returned 1 exit status

```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -139,7 +139,7 @@ Integration tests are composition of more than one class and test their interact
 
 To ensure that the provided testcode covers the productive code you can do a coverage scan with gcov. The reporting is done with lcov and htmlgen.
 You will need to install the following packages:
-```
+```bash
 sudo apt install lcov
 ```
 
@@ -148,16 +148,19 @@ The coverage scan applies to Quality level 3 and partly level 2 with branch cove
 
 For having a coverage report iceoryx needs to be compiled with coverage flags and the tests needs to be executed.
 You can do this with one command in iceroyx folder like this:
-```
+```bash
 ./tools/iceoryx_build_test.sh clean build-all -c <testlevel>
 ```
 Optionally you can use build-all option to get coverage for extensions like DDS or C-Binding.
 The -c flag indicates that you want to have a coverage report and you can pass there the needed testlevel. Per default the testlevel is set to 'all'.
 example:
+```bash
+./tools/iceoryx_build_test.sh debug build-all -c unit
 ```
-./tools/iceoryx_build_test.sh build-all clean -c unit
-```
-For having only reports for unit-test. In the script tools/gcov/lcov_generate.sh is the initial scan, filtering and report generation automatically done.
+**NOTE**
+Iceoryx needs to be build as static library for working with gcov flags. The script does it automatically.
+
+The flag `-c unit` is for having only reports for unit-tests. In the script `tools/gcov/lcov_generate.sh` is the initial scan, filtering and report generation automatically done.
 
 All reports are stored locally in build/lcov as html report (index.html). In Github we are using for codecov for a general reporting of the code coverage. 
 Codecov gives a brief overview over the code coverage and also indicates in Pull-Requests if new added code is not covered by tests.

--- a/doc/website/advanced/installation-guide-for-contributors.md
+++ b/doc/website/advanced/installation-guide-for-contributors.md
@@ -48,21 +48,24 @@ UndefinedBehaviorSanitizer (UBSan) is a fast undefined behavior detector. Iceory
 
 In iceoryx are scripts available to do the scan on your own. Additionally the Scans are running on the CI in every Pull-Request.
 As Requirement you should install the clang compiler:
-```
+```bash
 sudo apt install clang
 ```
 
 Then you need to compile the iceoryx with the sanitizer flags:
-```
+```bash
 ./tools/iceoryx_build_test.sh build-strict build-all sanitize clang clean
 ```
 After that we can run the tests with enabled sanitizer options:
-```
+```bash
 cd build
 ../tools/run_all_tests.sh
 ```
 When the tests are running without errors then it is fine, else an error report is shown with a stacktrace to find the place where the leak occurs. If the leak comes from an external dependency or shall be handled later then it is possible to set a function on a suppression list.
 This should be only rarely used and only in coordination with an iceoryx maintainer.
+
+**NOTE**
+Iceoryx needs to be build as static library for working with sanitizer flags. The script does it automatically.
 
 ## Iceoryx library build
 

--- a/tools/iceoryx_build_test.sh
+++ b/tools/iceoryx_build_test.sh
@@ -60,6 +60,7 @@ while (( "$#" )); do
         BUILD_TYPE="Debug"
         RUN_TEST=true
         COV_FLAG="ON"
+        BUILD_SHARED="OFF"
         if [ -z "$2" ]
         then
             shift 1
@@ -150,6 +151,7 @@ while (( "$#" )); do
         echo " [i] Build with sanitizers"
         BUILD_TYPE="Debug"
         SANITIZE_FLAG="ON"
+        BUILD_SHARED="OFF"
         shift 1
         ;;
     "clang")

